### ISSUE TITLE
feat(transforms): learned invertible leaky-ReLU (InvAutoAct)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#69]
## Problem / Request
Add a learned invertible pointwise nonlinearity: a generalised leaky-ReLU with per-channel positive slopes parameterised in log-space. Two variants — symmetric (single slope) and two-sided (separate positive/negative slopes).

## User Story
> As a user building normalizing flows, I want a cheap learned invertible activation I can insert between linear layers that isn't a spline or CDF, so that I can replicate INN / cINN architectures from Ardizzone et al. that interleave these with affine couplings.

## Motivation
- **Expressivity without splines.** Splines need bin counts + tails; a pointwise two-parameter activation is O(d) with trivial logdet.
- **Fills a gap.** gauss_flows has `InverseGaussCDF`, `HistogramCDF`, `MixtureGaussianCDF/LogisticCDF`, `RQSplineMarginal` — all heavy. A cheap piecewise-linear activation slots alongside.
- **Used in cINN / INN literature** — Ardizzone 2018, IB-INN, GIN.

## Proposed API
```python
from typing import ClassVar
from flowjax.bijections import AbstractBijection
from jax import Array

class InvAutoAct(AbstractBijection):
    """Learned symmetric invertible activation: y = x * exp(alpha).

    `alpha` is a per-dim log-slope. log|det J| = sum(alpha).
    """
    shape: tuple[int, ...]
    cond_shape: ClassVar[None] = None
    log_alpha: Array            # (d,)

class InvAutoActTwoSided(AbstractBijection):
    """Learned two-sided invertible leaky-ReLU.

        y = x * exp(alpha_pos)  for x >= 0
        y = x * exp(alpha_neg)  for x <  0

    Both slopes positive ⇒ strictly monotone ⇒ invertible.
    log|det J| = Σ_i  1[x_i>=0] * alpha_pos_i + 1[x_i<0] * alpha_neg_i
    """
    shape: tuple[int, ...]
    cond_shape: ClassVar[None] = None
    log_alpha_pos: Array        # (d,)
    log_alpha_neg: Array        # (d,)
```

## Design Snapshot
```python
# src/gauss_flows/_src/transforms/bijections/elementwise/inv_auto_act.py
import jax.numpy as jnp
from typing import ClassVar
from flowjax.bijections import AbstractBijection
from jax import Array

class InvAutoActTwoSided(AbstractBijection):
    shape: tuple[int, ...]
    cond_shape: ClassVar[None] = None
    log_alpha_pos: Array
    log_alpha_neg: Array

    def __init__(self, shape: tuple[int, ...]):
        self.shape = shape
        self.log_alpha_pos = jnp.zeros(shape)
        self.log_alpha_neg = jnp.zeros(shape)

    def transform_and_log_det(self, x, condition=None):
        x = jnp.asarray(x)
        mask = x >= 0
        log_slope = jnp.where(mask, self.log_alpha_pos, self.log_alpha_neg)
        y = x * jnp.exp(log_slope)
        return y, jnp.sum(log_slope)

    def inverse_and_log_det(self, y, condition=None):
        y = jnp.asarray(y)
        # sign(y) == sign(x) because both slopes > 0; branch on sign(y)
        mask = y >= 0
        log_slope = jnp.where(mask, self.log_alpha_pos, self.log_alpha_neg)
        x = y * jnp.exp(-log_slope)
        return x, -jnp.sum(log_slope)
```

## Mathematical Notes
```text
Forward (per channel i):
    y_i = x_i * exp(α_i^+)   if x_i >= 0
    y_i = x_i * exp(α_i^-)   if x_i  < 0

Since both slopes are positive, sign(y_i) = sign(x_i), so the inverse
indexes the branch by sign(y) — no fixed-point solver needed.

log|det J| = Σ_i 1[x_i >= 0] α_i^+ + 1[x_i < 0] α_i^-

Edge case: x_i = 0 ⇒ y_i = 0 for either branch; convention: x >= 0 uses α^+.
```

## References & Existing Code
- Paper: Ardizzone et al. 2018. *Analyzing Inverse Problems with Invertible Neural Networks*. arXiv:1808.04730
- Reference impl: `vislearn/FrEIA/FrEIA/modules/inv_auto_layers.py::InvAutoAct`, `InvAutoActTwoSided`
- gauss_flows analogue: none — novel addition

## Implementation Steps
- [ ] Add `InvAutoAct` + `InvAutoActTwoSided` in `src/gauss_flows/_src/transforms/bijections/elementwise/inv_auto_act.py`
- [ ] Re-export via elementwise `__init__.py`, bijections `__init__.py`, transforms `__init__.py`, top-level `gauss_flows`
- [ ] Add `__all__` entries

## Definition of Done
- [ ] Code at intended path
- [ ] Public API exported
- [ ] Tests pass
- [ ] Lint + typecheck pass
- [ ] Google-style docstrings

## Testing
- [ ] Round-trip: `inverse(transform(x)) == x` to machine precision
- [ ] Log-det consistency: `fwd_log_det + inv_log_det == 0`
- [ ] At default init (log_alpha = 0): identity map, `log_det == 0`
- [ ] Sign-branch correctness on handcrafted inputs crossing zero
- [ ] `InvAutoAct` (symmetric) matches two-sided when `alpha_pos == alpha_neg`

## Documentation
- [ ] API reference entry

## Relationships
- Related: FrEIA port series
